library/standard_tables_capture.inc

```diff
@@ -243,6 +244,144 @@ function snomed_import($us_extension=FALSE) {
+    if (!$us_extension) {
+        //var_dump($us_extension);
+        foreach($table_array_for_snomed as $val){
+            if(trim($val)!=''){
+                sqlStatement($val);
+            }
+        }
+    }
+
+    // reading the SNOMED directory and identifying the files to import and replacing the variables by originals values.
+    if( is_dir($dir) && $handle = opendir($dir)) {
+        while (false !== ($filename = readdir($handle))) {
+            if ($filename != "." && $filename != ".." && !strpos($filename,"zip")) {
+                $path=$dir."".$filename."/".$sub_path;
+                if (!(is_dir($path))) {
+                    $path=$dir."".$filename."/RF2Release/".$sub_path;
+                }
+                if( is_dir($path) && $handle1 = opendir($path)) {
+                    while (false !== ($filename1 = readdir($handle1))) {
+                        $load_script="Load data local infile '#FILENAME#' into table #TABLE# fields terminated by '\\t' ESCAPED BY '' lines terminated by '\\n' ignore 1 lines   ";
+                        $array_replace=array("#FILENAME#","#TABLE#");
+                        if ($filename1 != "." && $filename1 != "..") {
+                            $file_replace=$path.$filename1;
+                            if(strpos($filename1,"Concept") !== false){
+                                $new_str=str_replace($array_replace,array($file_replace,"sct2_concept"),$load_script);
+                            }	
+                            if(strpos($filename1,"Description") !== false){
+                                $new_str=str_replace($array_replace,array($file_replace,"sct2_description"),$load_script);
+                            }
+                            if(strpos($filename1,"Identifier") !== false){
+                                $new_str=str_replace($array_replace,array($file_replace,"sct2_identifier"),$load_script);
+                            }
+                            if(strpos($filename1,"Relationship") !== false){
+                                $new_str=str_replace($array_replace,array($file_replace,"sct2_relationship"),$load_script);
+                            }  
+                            if(strpos($filename1,"StatedRelationship") !== false){
+                                $new_str=str_replace($array_replace,array($file_replace,"sct2_statedrelationship"),$load_script);
+                            }  
+                            if(strpos($filename1,"TextDefinition") !== false){
+                                $new_str=str_replace($array_replace,array($file_replace,"sct2_textdefinition"),$load_script);
+                            }  
+                            if($new_str!=''){
+                                sqlStatement($new_str);
+                            }
+                        }
+                    }
+                }
+                closedir($handle1);
+            }
+        }
+        closedir($handle);
+    }
```

Comment:
If this is successful, then should do 2 more things:
1. truncate the old sct tables if the exist.
2. In the code_types table, for the snomed items, change the ct_external fields to the corresponding rf2 items.

(and should also add the same opposite actions when installing a rf1 snomed package)

(This means that when changing to or from rf2, the old tables will get truncated to save space and the connectors will be automatically modified to point to the new tables)